### ISSUE TITLE
Easter Egg Customization

### ISSUE THREAD

**Issue by bkimminich** [juice-shop/juice-shop#459]
Make the following customizable via `config/*.yml` properties:
* `planetOverlayMap` can be a file or URL (which is downloaded on startup) that is rendered as a map on the easter egg 3D planet (default: `orangemap2k.jpg`)
* `planetName` can be a String overwriting the default name of the easter egg 3D planet in the `<title>Welcome to Planet ____________</title>` of `threejs-demo.html` (default: `Orangeuze`)

**Comment by WVik** [comment]
Hello @bkimminich . I had gone through the codebase a week ago. Most of it is clear to me, but there's a little thing I can't figure out related to this issue. I figured out how I would proceed with the issue. What is unclear to me is how are the AngularJS views communicating with the config yml files? I inspected the Models and Controllers, but couldn't find it. Sorry for my ignorance, have never worked with yaml before.

**Comment by bkimminich** [comment]
The AngularJS client uses the https://github.com/bkimminich/juice-shop/blob/master/app/js/services/ConfigurationService.js to retrieve the configuration as a JSON object: `{ config: { server: { port: 3000 }, application: { ... } } }`. Check out http://juice-shop-staging.herokuapp.com/rest/admin/application-configuration to see how the `default.yml` looks like when returned through the REST API.

**Comment by WVik** [comment]
Thanks @bkimminich ! I figured out how it works now. Now since `threejs-demo.html` doesn't have an Angular controller as there's no session related data to be displayed on it, should I use a xhr request to get the filename (image name from the JSON)?

**Comment by bkimminich** [comment]
Hm, thinking about it, there is a much easier way, identical to how it's done in the `index.template.html`: https://github.com/bkimminich/juice-shop/blob/master/server.js#L271-L308

Making a similar `populateThreeJsTemplate()` function where title and that image path are replaced using the `replace` function, would do the trick.

**Comment by shoebpate1** [comment]
Addressed in #460

**Comment by WVik** [comment]
@bkimminich  Thanks again! That's a nice way of doing it. I was wondering why create a separate `threejs-demo-template.html` file for incorporating changes when we can use `fs.open` in `wx` mode to modify the existing `threejs-demo.html` file? I checked locally and it is working fine. Is there a particular reason for doing it this way?

**Comment by bkimminich** [comment]
~~Tbh, I forgot why I did this in the first place... Might be a good follow-up refactoring to apply KISS principle here... 😂~~

I just remembered why I did this! :grin: When changing the existing file, it will very likely slip into Git at some point in a changed state. With the templates in Git and the copied actual files in `.gitignore` this cannot happen. For the same reason I also listed all product image filenames that belong to one of the included alternate themes in `.gitignore`.

So, I think this is it for this task then! :+1:
